gdk/common/RecipeValidator.py

```diff
@@ -260,3 +260,75 @@ def _convert_keys_to_lowercase(self, input_dict):
             return [self._convert_keys_to_lowercase(item) for item in input_dict]
         else:
             return input_dict
+
+    def _convert_keys_to_camelcase(self, input_data):
+        """
+        Recursively converts keys in a nested dictionary to CamelCase based on the predefined mapping.
+
+        Parameters
+        ----------
+        input_data : CaseInsensitiveDict or list
+            The input dictionary or list to be converted.
+
+        Returns
+        -------
+        CaseInsensitiveDict or list
+            The input dictionary or list with keys converted to CamelCase.
+
+        """
+        if isinstance(input_data, CaseInsensitiveDict):
+            result_dict = {}
+            for key, value in input_data.items():
+                camelcase_key = self.RECIPE_PROPERTY_CASE_MAPPING.get(key.lower(), key)
+                result_dict[camelcase_key] = self._convert_keys_to_camelcase(value)
+            return result_dict
+        elif isinstance(input_data, list):
+            result_list = []
+            for item in input_data:
+                result_list.append(self._convert_keys_to_camelcase(item))
+            return result_list
+        else:
+            return input_data
+
+    RECIPE_PROPERTY_CASE_MAPPING = {
+        "recipeformatversion": "RecipeFormatVersion",
```

Comment:
Can we generate this from the recipe schema, rather than re-defining these here? I think it should be possible, by reading all properties from the schema (which would get the camelcase data), and then converting that to lowercase and assembling it into a map like this. This would make maintenance on the schema easier, by not needing to also update this.